jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/JibExtension.java

```diff
@@ -221,6 +227,17 @@ public String getContainerizingMode() {
     return property != null ? property : containerizingMode.get();
   }
 
+  @Input
+  @Optional
+  public Property<String> getConfigurationName() {
+    return configurationName;
```

Comment:
our of curiosity, can we just set the property to the system property value here?
```
public Property<String> getConfigurationName() {
  String property = System.getProperty(PropertyNames.CONFIGURATION_NAME);
  if (property != null) {
    configurationName.set(property);
  }
  return configurationName;
}
```

while this still looks a little odd, if it is possible, it matches more closely our previous patterns and it removes the ambiguous `GetconfigurationName/readConfigurationName` situation.

Reply:
I've realized it via a default value now, that sounds like a good idea anyway because it gives the user all the control. So if you want the override behavior you can just set the value to `System.getProperty("some.property", "default")`. The downside is of course that this is different from the rest of the API, creating an inconsistency in some cases.